packages/java/endpoint/src/main/java/dev/hilla/crud/PropertyStringFilterSpecification.java

```diff
@@ -75,6 +79,56 @@ public Predicate toPredicate(Root<T> root, CriteriaQuery<?> query,
             default:
                 break;
             }
+        } else if (isLocalDate(javaType)) {
+            var path = root.<LocalDate> get(filter.getPropertyId());
+            var dateValue = LocalDate.parse(value);
+            switch (filter.getMatcher()) {
+            case EQUALS:
+                return criteriaBuilder.equal(path, dateValue);
+            case CONTAINS:
+                throw new IllegalArgumentException(
+                        "A date cannot be filtered using contains");
+            case GREATER_THAN:
+                return criteriaBuilder.greaterThan(path, dateValue);
+            case LESS_THAN:
+                return criteriaBuilder.lessThan(path, dateValue);
```

Comment:
For many real-world applications, the operator "between" or a combination of greater_than and less_than is the default for filtering date values. Also, it is vague that whether these two are inclusive or exclusive.

Reply:
That could be improved, through another issue

Reply:
We're talking about a header filter, there's no room for complex UI bits there

Reply:
I'm not sure having support for between is a luxury just because it is header filtering. In fact, by default it should show a special control in header title not just a DatePicker (or a DateTimePicker) which by clicking on that it shows a popup containing two DatePicker (or the other variants). Then based on whether the user has chosen a value for either one or both of them, it should do the filtering by greater_than, less_than or between. This can be a future improvement of course, but it is not something extra ordinary.

Reply:
But, still the answer to the other question is not clear, how the user or even the developer can indicate whether it is performing a greater_than (exclusive) or greater_than_or_equals_to (inclusive). Should it be indicated with a sign, icon, tooltip, etc.?

Reply:
For the between case, I created this: https://github.com/vaadin/hilla/issues/1463